File: medium/1291_sequential_digits/solution.cpp
#include <vector>

class Solution {
public:
  std::vector<int> sequentialDigits(int low, int high) {
    const int digits[9] = {1, 2, 3, 4, 5, 6, 7, 8, 9};

    std::vector<int> sequentialNums;

    for (size_t count = 1; count < 10; ++count) {
      for (size_t i = 0; i < 9; ++i) {
        if (i == 10 - count)
          break;

        int sequentialNum = 0;

        for (size_t j = 0; j < count; ++j)
          sequentialNum = sequentialNum * 10 + digits[i + j];

        if (low <= sequentialNum && sequentialNum <= high)
          sequentialNums.push_back(sequentialNum);
      }
    }

    return sequentialNums;
  }
};

File: medium/1291_sequential_digits/solution.py
from typing import List


class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        lowStr = str(low)

        seed = ord(lowStr[0]) - 48
        sLen = len(lowStr)

        sequentialNums = []

        while True:
            sequentialNum = 0
            for i in range(sLen):
                sequentialNum = sequentialNum * 10 + seed + i

            if sequentialNum > high:
                break
            elif seed < sequentialNum % 10 and sequentialNum >= low:
                sequentialNums.append(sequentialNum)

            if sequentialNum % 10 == 9:
                seed = 1
                sLen += 1
            else:
                seed += 1

        return sequentialNums
